### PR TITLE
scala3: revert some backward incompat changes

### PR DESCRIPTION
**Pull request by tnielens**
This PR into pityka/feature/scala3 improves two things in the base branch:
- rework the ScalarTag to avoid the binary incompatibility
- avoid introducing `naToInt` and `naInt`

### CODE REVIEW

**Comment by pityka** [comment]
Thank you! I will read it over the weekend and if it is compatible then there is no reason not to merge the whole thing.

File: saddle-core/src/main/scala/org/saddle/scalar/ScalarTag.scala [hunk 1]
```diff
@@ -53,42 +53,63 @@ trait ScalarTag[@spec(Boolean, Int, Long, Float, Double) T]
...
 
+  override def erasure = runtimeClass
+
   // forward 2.10 compatibility
   def runtimeClass: Class[_]
 }
 
 object ScalarTag extends ScalarTagImplicits {
-  implicit def scalarTagToClassTag[T](implicit st: ST[T]) : ClassTag[T] = st.classTag
+  val scalarTagToClassTag = 1
```
Comment (pityka) [inline saddle-core/src/main/scala/org/saddle/scalar/ScalarTag.scala:80]: do you need this?

Comment (tnielens) [reply saddle-core/src/main/scala/org/saddle/scalar/ScalarTag.scala:80]: No idd, left over from the removal.

**Comment by tnielens** [comment]
Actually, a lot of this PR is a revert. To avoid unnecessary diffs and code annotations, don't hesitate to squash this commit if you merge or cherry pick it.

**Pull request merged**